package_metrics/datadog_utils.py

```diff
@@ -0,0 +1,64 @@
+import json
+import os
+import time
+
+import requests
+
+from package_metrics.constants import (
+    DATADOG_API_KEY,
+    DATADOG_APP_KEY,
+    GITHUB_ACTIONS,
+    NOSE_DIVIDED_WE_RUN,
+    MetricType,
+)
+
+
+def send_metric(name, value, metric_type, tags=None):
+    """
+    Send a specific metric to datadog
+    :param name: name of metric
+    :param value:
+    :param metric_type: str in constants.MetricType
+    :param tags: optional {key: value}, append additional tags to metric
+           e.g., {"tag_name": "tag_value"}
+    """
+
+    api_key = os.environ[DATADOG_API_KEY]
+    app_key = os.environ[DATADOG_APP_KEY]
+
+    if metric_type not in MetricType.options():
+        raise ValueError(
+            f"Expected {metric_type} to be one of the following: "
+            f"{MetricType.options()}")
+
+    if GITHUB_ACTIONS in os.environ and os.environ[GITHUB_ACTIONS]:
+        host = 'github.com'
+        ci_env = 'github_actions'
+    else:
+        host = 'unknown'
+        ci_env = 'unknown'
+
+    current_time = int(time.time())  # drop milliseconds from time.time()
+
+    url = 'https://app.datadoghq.com/api/v1/series'
+    headers = {
+        'Content-type': 'application/json',
+        'DD-API-KEY': api_key,
+        'DD-APP-KEY': app_key,
+    }
+
+    tags = tags or {}
+    tags['environment'] = ci_env
+    if NOSE_DIVIDED_WE_RUN in os.environ and os.environ[NOSE_DIVIDED_WE_RUN]:
+        tags['partition'] = os.environ[NOSE_DIVIDED_WE_RUN]
+
+    payload = {
+        'series': [{
+            'metric': name,
+            'points': [[current_time, value]],
+            'type': metric_type,
+            'host': host,
+            'tags': [tags]
```

Comment:
This looks different than what was done by `scripts/datadog-utils.sh`. Is it acceptable to pass a list containing a dict rather than a list of `"<tag>:<value>"` strings?

Reply:
Good catch! Looking closer at [the docs](https://docs.datadoghq.com/api/latest/metrics/#submit-metrics), it definitely expects a list of strings. Updated to match `scipts/datadog-utils.sh` in 1a6ac1886e7128370aa042787ce46b11e628e48f